### PR TITLE
[PROD-899] report library name and version in X-Pusher-Library header

### PR DESCRIPTION
**Pull request by jameshfisher**

### CODE REVIEW

File: lib/requests.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ var request = require('request');
 var errors = require('./errors');
 var util = require('./util');
 
+var packageJson = require('../package.json');
```
Comment (topliceanu) [inline lib/requests.js]: I don't think you need the `.json` extension

File: lib/requests.js [hunk 2]
```diff
@@ -27,14 +29,17 @@ function send(config, options, callback) {
...
+    headers: {
+      'x-pusher-library': 'pusher-http-node ' + packageJson.version
+    }
   };
 
   if(body) {
-  	params.headers = {'content-type': 'application/json'};
-  	params.body = body;
+    params.headers['content-type'] = 'application/json';
+    params.body = body;
```
Comment (topliceanu) [inline lib/requests.js]: Also note the whitespace inconsistencies, I think this project defaults to 4 white spaces per tab, your editor is set to 2. In some parts of the world this is considered criminal!

Comment (jameshfisher) [inline lib/requests.js]: Haha :)

This file at least is 4 spaces everywhere except for those lines which I corrected, which had an errant tab character in them

File: lib/requests.js (hunk 1 above)
Comment (jameshfisher) [inline lib/requests.js]: You're right!

**Pull request merged**